server/app/services/geo/esri/FakeEsriClient.java

```diff
@@ -0,0 +1,155 @@
+package services.geo.esri;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.time.Clock;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.libs.Json;
+import services.AddressField;
+import services.geo.AddressLocation;
+
+/**
+ * Provides overide methods for {@link EsriClient} to illiminate the need for calling the external
+ * Esri services for dev and testing.
+ */
+public final class FakeEsriClient extends EsriClient {
+  private String jsonResources = System.getProperty("user.dir") + "/test/resources/esri/";
+  private JsonNode addressCandidates;
+  private JsonNode noAddressCandidates;
+  private JsonNode serviceAreaFeatures;
+  private JsonNode serviceAreaNoFeatures;
+  private JsonNode serviceAreaNotInArea;
```

Comment:
**issue (blocking):** Ensure these members won't be reused between requests.

By default, Play uses dependency injection to instantiate an instance of the controller needed to handle a request, and then reuses that instance for subsequent requests. Dependencies of that controller (and therefore their dependencies) will similarly be reused. There are two options for changing this behavior, one is to inject a `Provider<YouClass>` and call `get` on it to force creation of a new instance. The other is to prefix the route with an `@` in the routes file, forcing Play to inject a `Provider<YourController>` in its generated router class.

More info here: https://www.playframework.com/documentation/2.8.x/JavaDependencyInjection#Dependency-injecting-controllers

Reply:
If I'm reading this correctly (and I may not be), I think it's okay for these to be reused.  They will contain the contents of the json files that are used for the mock data, and it's just a caching mechanism for that.  None of that data will change.